app/controllers/form_controller.rb

```diff
@@ -100,9 +100,10 @@ def restore_error_field_values(questions)
     return unless questions
 
     questions.each do |question|
-      if question&.type == "date" && @log.attributes.key?(question.id)
-        @log[question.id] = @log.send("#{question.id}_was")
-      end
+      next unless question&.type == "date" && @log.attributes.key?(question.id)
+      next unless @log[question.id]&.year&.zero?
+
+      @log[question.id] = @log.send("#{question.id}_was")
```

Comment:
The reason we reset the date to what it was is because when it's an invalid date format, it gets reset to day: 1, month: 0, year: 0

We don't have to reset it if the format is valid to begin with, which this does. 

The reason this is relevant for compound validations is because when we click "see all related answers" we resubmit the form with whatever is entered for the questions on the current page in order to get all the validation errors.

If the date gets reset, we end up resubmitting a valid date, so there are no longer any validation errors and the list with error fields shows up empty.